src/flint/flintlib/nmod_poly_factor.pxd

```diff
@@ -6,9 +6,9 @@ cdef extern from "flint/nmod_poly_factor.h":
 
     ctypedef struct nmod_poly_factor_struct:
         nmod_poly_struct *p
-        long *exp
-        long num
-        long alloc
+        slong *exp
+        slong num
+        slong alloc
```

Comment:
This was out of data compared to what I see in the Flint code.

On 64-bit Windows at least `long` and `slong` are not equivalent and this has been a common cause of Windows-specific crashes.

I think over time Flint itself has cleaned up its use of slong in place of long for many interfaces. Some of python-flint's code predates Flint cleaning these things up though so we have some things like this that are out of date.

It is likely that almost all of these 635 lines should change from long to slong:
```console
$ git grep ' long ' | wc -l
635
```
Certainly these are out of date because `prec` should be `slong`:
```console
$ git grep long src/flint/flintlib/arb.pxd | head
src/flint/flintlib/arb.pxd:from flint.flintlib.flint cimport ulong, slong, flint_rand_t
src/flint/flintlib/arb.pxd:    void arb_set_round(arb_t z, const arb_t x, long prec)
src/flint/flintlib/arb.pxd:    void arb_neg_round(arb_t x, const arb_t y, long prec)
src/flint/flintlib/arb.pxd:    void arb_set_si(arb_t x, long y)
src/flint/flintlib/arb.pxd:    void arb_set_ui(arb_t x, ulong y)
src/flint/flintlib/arb.pxd:    void arb_set_round_fmpz_2exp(arb_t y, const fmpz_t x, const fmpz_t exp, long prec)
src/flint/flintlib/arb.pxd:    void arb_set_round_fmpz(arb_t y, const fmpz_t x, long prec)
src/flint/flintlib/arb.pxd:    void arb_printd(const arb_t x, long digits)
src/flint/flintlib/arb.pxd:    void arb_mul_2exp_si(arb_t y, const arb_t x, long e)
src/flint/flintlib/arb.pxd:    void arb_get_abs_ubound_arf(arf_t u, const arb_t x, long prec)
```
https://flintlib.org/doc/arb.html